src/rgw/rgw_lc.cc

```diff
@@ -1445,7 +1460,7 @@ class LCOpAction_Transition : public LCOpAction {
       // send request to notification manager
       int publish_ret =  notify->publish_commit(oc.dpp, obj_state->size,
 				    ceph::real_clock::now(),
-				    obj_state->attrset[RGW_ATTR_ETAG].to_str(),
+				    etag,
```

Comment:
sorrry, just noticed when reviewing https://github.com/ceph/ceph/pull/57079 - this call to `publish_commit()` still relies on `obj_state->size`. don't we have the same lifetime issue there?

Reply:
Right and moreover post cloud-transition obj_size can be '0' which can be misleading in the notification.

Reply:
right, we should save the value of size before the expiration/transition like you did for etag

what do you want to do about the downstream version of this change?